### ISSUE TITLE
Simplify config package usage

### ISSUE THREAD

**Issue by tbalthazar** [buildpack/pack#281]
The `config` package has a [`Write`](https://github.com/buildpack/pack/blob/master/config/config.go#L51) method that allows to write to paths outside of `PACK_HOME`.

I don't know if there's a use case for that, but what I would be tempted to do is:
- scope the method to `PACK_HOME`
  - `Write(cfg, "foo.toml")` would write to `PACK_HOME/foo.toml`
  - `Write(cfg, "bar/foo.toml")` would write to `PACK_HOME/bar/foo.toml`

I also noticed that this method is only called twice, writing to `config.DefaultConfigPath` each time. I suspect that this will be the most common use case, so I would be tempted to add a new method in the `config` package:

```go
func WriteDefaultConfig(cfg Config) error {
  // call Write with DefaultConfigPath
}
```

It would remove boilerplate code here and there.

`config.Read` is called only once and again with `config.DefaultConfigPath`. A new helper method might be handy too:

```go
func DefaultConfig() (Config, error) {
  // call Read with DefaultConfigPath
}
```

All that leads me to wonder if we still need `Read(path string)` and `Write(cfg Config, path string)` at all? Are there anticipated use cases to read/write config that isn't the Default one? (pretty new to the project, sorry if it's a naïve question)

So this is mainly me thinking out loud here, let me know if that makes sense or not, or if it is simply not a priority for the project right now.

If you think it would be helpful, I'd be happy to start working on a PR.

(more context in the conversations with @jromero in https://github.com/buildpack/pack/pull/275)

**Comment by ekcasey** [comment]
@tbalthazar we want the functionality in `pack` to be consumable by other tools as a library in addition to being useful as a CLI. I think keeping the config path as a parameter allows flexibility for CLI tools that may want to import functionality from `pack` but don't want the environment variable `PACK_HOME` to be part of their interface.

**Comment by tbalthazar** [comment]
Makes sense, closing this then 👍
